### ISSUE TITLE
IndexError: string index out of range on reposition_match

### ISSUE THREAD

**Issue by cixtor** [SublimeLinter/SublimeLinter#1028]
### Steps to Reproduce the Exception

* Create a new file `foo.php`
* PHP_CodeSniffer version 3.2.2
* Enable "Word Wrap" in the whole file
* [SublimeLinter-phpcs](https://github.com/SublimeLinter/SublimeLinter-phpcs) with PSR2
* Wait until the linter flags the errors
* Place the cursor in line 9 column 33 before `"Lorem`
* Select the code until line 9 column 481 before `$bar"`
* With the code selected, quickly type `user(`
* The exception panel automatically appears

### Full Copy of the Exception

```
Traceback (most recent call last):
  File "/SublimeLinter/lint/backend.py", line 201, in await_futures
    yield future.result()
  File "./python3.3/concurrent/futures/_base.py", line 394, in result
  File "./python3.3/concurrent/futures/_base.py", line 353, in __get_result
  File "./python3.3/concurrent/futures/thread.py", line 54, in run
  File "/SublimeLinter/lint/backend.py", line 96, in execute_lint_task
    errors = linter.lint(code, view_has_changed, settings) or []
  File "/SublimeLinter/lint/linter.py", line 838, in lint
    error = self.process_match(m, vv)
  File "/SublimeLinter/lint/linter.py", line 914, in process_match
    line, start, end = self.reposition_match(m.line, col, m, vv)
  File "/SublimeLinter/lint/linter.py", line 959, in reposition_match
    if near[pos].isalnum() or near[pos] == '_':
IndexError: string index out of range
```

### Errors Generated By The Linter

```
foo.php:
 9:9   error  phpcs: error  Opening parenthesis of a multi-line function call must be the last content on the line
 9:33  error  phpcs: error  Only one argument is allowed per line in a multi-line function call
10:1   error  phpcs: error  Multi-line function call not indented correctly; expected 8 spaces but found 12
10:34  error  phpcs: error  Closing parenthesis of a multi-line function call must be on a line by itself
```

### Code Snippet To Reproduce The Bug

```
<?php

function abc()
{
    /* Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. */
    if ($a == 1 && $b == 2) {
        $msg = str_replace("/path/to/a/random/directory/path/", "/path/to/another/random/directory/path/", $foo);
        echo "LOREM IPSUM DOLOR: $foo\n";
        file_put_contents($msg, "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum: $bar",
            FILE_APPEND | LOCK_EX);
    }
}
```

### Animation Showing The Exception

![SublimeLinter IndexError Exception](https://user-images.githubusercontent.com/1933999/36759931-d5389b4c-1bcd-11e8-9d59-7057173e2427.gif)

**Comment by kaste** [comment]
What was the lint error here? Was the lint error correct?

**Comment by cixtor** [comment]
The error was correct, it was a generic PHP syntax error.

This exception was triggered when a portion of the code in the same line as the warning was selected and immediately replaced with another piece of code. I do not understand the logic of the code that is powering this part of the linter, but taking the name of the `reposition_match` method it makes sense, considering the conditions _(modifying the code after the warning is drawn)_.

**Comment by kaste** [comment]
Can you post the lint error, and which linter.

**Comment by cixtor** [comment]
I have updated the description of the original issue with the PoC and a GIF.

**Comment by kaste** [comment]
Possible fix in #1031
